src/ziggurat/messaging/consumer.clj

```diff
@@ -39,45 +39,81 @@
       (lb/reject ch delivery-tag false)
       nil)))
 
-(defn- try-consuming-dead-set-messages [ch ack? queue-name topic-entity]
+(defn convert-message
+  "Take the ch metadata payload and ack? as parameter.
+   Decodes the payload the ack it if ack is enabled and returns the message"
+  [ch delivery-tag ^bytes payload topic-entity]
+  (try
+    (let [message (nippy/thaw payload)]
+      (convert-to-message-payload message topic-entity))
+    (catch Exception e
+      (sentry/report-error sentry-reporter e "Error while decoding message")
+      (lb/reject ch delivery-tag false)
+      nil)))
+
+(defn- ack-message
+  ([ch delivery-tag]
+   (ack-message ch delivery-tag true))
+  ([ch delivery-tag ack?]
+   (when ack?
+     (lb/ack ch delivery-tag))))
+
+(defn process-message [ch meta payload topic-entity processing-fn]
```

Comment:
in the docs, we should mention that the processing fn should have an arity of 1 and that will be a clojure map